src/containers/TriggerTemplate/TriggerTemplate.js

```diff
@@ -0,0 +1,386 @@
+      />
+    );
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { match, webSocketConnected } = this.props;
+    const { namespace, triggerTemplateName } = match.params;
+    const {
+      match: prevMatch,
+      webSocketConnected: prevWebSocketConnected
+    } = prevProps;
+    const {
+      namespace: prevNamespace,
+      triggerTemplateName: prevTriggerTemplateName
+    } = prevMatch.params;
+
+    if (
+      namespace !== prevNamespace ||
+      triggerTemplateName !== prevTriggerTemplateName ||
+      (webSocketConnected && prevWebSocketConnected === false)
+    ) {
+      this.fetchData();
+    }
+  }
+
+  fetchData() {
+    const { match } = this.props;
+    const { namespace, triggerTemplateName } = match.params;
+    this.props.fetchTriggerTemplate({ name: triggerTemplateName, namespace });
+  }
+
+  render() {
+    const {
+      intl,
+      error,
+      match,
+      selectedNamespace,
+      triggerTemplate
+    } = this.props;
+
+    if (error) {
+      return TriggerTemplateContainer.notification({
+        kind: 'error',
+        intl
+      });
+    }
+    if (!triggerTemplate) {
+      return TriggerTemplateContainer.notification({
+        kind: 'error',
+        message: 'TriggerTemplate not available',
+        intl
+      });
+    }
+
+    const { params, resourcetemplates } = triggerTemplate.spec;
+    const { triggerTemplateName } = match.params;
+    const triggerTemplateDate = triggerTemplate.metadata.creationTimestamp;
+    const triggerTemplateNamespace = triggerTemplate.metadata.namespace;
+
+    /* 
+      Labels: ${"baz":"boo","foo":"bar"} 
+      should become 
+      Labels:
+       foo: bar
+       baz: boo
+    */
+
+    let triggerTemplateLabels = '';
+    const formattedLabelsToRender = [];
+    if (triggerTemplate.metadata.labels) {
+      triggerTemplateLabels = JSON.stringify(triggerTemplate.metadata.labels)
+        .replace('{', '')
+        .replace('}', '')
+        .replace(/['"]+/g, '')
+        .replace('$', '');
+
+      const templateLabelsSplitOnComma = triggerTemplateLabels.split(',');
+      templateLabelsSplitOnComma.forEach(item => {
+        const splitOnColon = item.split(':');
+        formattedLabelsToRender.push(`  ${splitOnColon[0]}: ${splitOnColon[1]}
+`);
+      });
+    }
+
+    const headersForParameters = [
+      {
+        key: 'name',
+        header: intl.formatMessage({
+          id: 'dashboard.triggerTemplate.name',
+          defaultMessage: 'Name'
+        })
+      },
+      {
+        key: 'default',
+        header: intl.formatMessage({
+          id: 'dashboard.triggerTemplate.default',
+          defaultMessage: 'Default'
+        })
+      },
+      {
+        key: 'description',
+        header: intl.formatMessage({
+          id: 'dashboard.triggerTemplate.description',
+          defaultMessage: 'Description'
+        })
+      }
+    ];
+
+    let rowsForParameters = [];
+
+    if (params) {
```

Comment:
I feel this `if statement` is unnecessary given that if no params are available, when doing the map below will just result in `[]` just like it was initialized which will produce an empty table. My suggestion would be maybe to this if statement inside the render method to logically render the Param table if and only if there are params in the TriggerTemplate. Same way your doing for the `resourcetemplates`.

Reply:
I thought so too, but we end up mapping on undefined otherwise if there are no params 🤔